docs/api/selector.md

```diff
@@ -12,6 +12,16 @@ follows:
 - class syntax (`.foo`, `.foo-bar`, etc.)
 - tag syntax (`input`, `div`, `span`, etc.)
 - id syntax (`#foo`, `#foo-bar`, etc.)
+- attr syntax(`[foo="bar"]`, `[foo]`, etc.);
+
+A note on attribute selectors: For compatibilty, you can use either the
+react prop name, or the native html selector. For example, the following
```

Comment:
cc @lelandrichardson 

I'm open for changing either way.
@ljharb what about the context of different rendering layers. Think ReactNative, ReactHardware. In those contexts, do we just say the `[]` isn't for them? I think leland had a good point that the `[]` selector was built for XML, not explicitly the DOM. so in a ReactNative/Hardware standpoint, the Props *are* the attributes of the "rendered" xml